mock_ros2_project/action_client.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""
+ROS2 Action Client Example
+
+This node demonstrates how to create an action client that calls
+the Fibonacci action and handles feedback.
+"""
+
+import sys
+from typing import Optional
+
+import rclpy
+from rclpy.action import ActionClient
+from rclpy.node import Node
+from example_interfaces.action import Fibonacci
+
+
+class ActionClientNode(Node):
+    """Action client node that requests Fibonacci sequences."""
+    
+    def __init__(self) -> None:
+        """Initialize the action client node."""
+        super().__init__('action_client')
+        
+        # Create action client for Fibonacci action
+        self._action_client = ActionClient(self, Fibonacci, 'fibonacci')
+
+    def send_goal(self, order: int) -> None:
+        """Send a goal to the action server.
+        
+        Args:
+            order: The order of the Fibonacci sequence to generate
+        """
+        goal_msg = Fibonacci.Goal()
+        goal_msg.order = order
+        
+        if not self._action_client.wait_for_server(timeout_sec=5.0):
+            self.get_logger().error('Action server not available after waiting')
+            return
+        
```

Comment:
When the action server is unavailable you log the error and return, but the caller still proceeds to `rclpy.spin(action_client_node)` (S111–S113).  Consider shutting the node down here or propagating an exception so that the main loop can exit cleanly instead of spinning indefinitely without an active goal.

```suggestion
        if not self._action_client.wait_for_server(timeout_sec=5.0):
            self.get_logger().error('Action server not available after waiting')
            rclpy.shutdown()
            return
```